File: Python/1-bit and 2-bit Characters.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(1)
    """

    def isOneBitCharacter(self, bits: List[int]) -> bool:
        i, n = 0, len(bits) - 1

        while i < n:
            i += 2 if bits[i] == 1 else 1

        return i == n

File: Java/1-bit and 2-bit Characters.java
public class Solution {

    /**
     * Time:   O(n)
     * Memory: O(1)
     */
    public boolean isOneBitCharacter(int[] bits) {
        int i = 0, n = bits.length - 1;

        while (i < n) {
            i += bits[i] == 1 ? 2 : 1;
        }

        return i == n;
    }
}
